servicetalk-grpc-health/src/test/java/io/servicetalk/grpc/health/DefaultHealthServiceTest.java

```diff
@@ -0,0 +1,153 @@
+            }
+        }
+    }
+
+    @Test
+    void defaultWatch() throws Exception {
+        DefaultHealthService service = new DefaultHealthService();
+        try (ServerContext serverCtx = GrpcServers.forAddress(localAddress(0)).listenAndAwait(service)) {
+            try (Health.BlockingHealthClient client = GrpcClients.forResolvedAddress(
+                    (InetSocketAddress) serverCtx.listenAddress()).buildBlocking(new Health.ClientFactory())) {
+                BlockingIterator<HealthCheckResponse> itr = client.watch(newRequest(OVERALL_SERVICE_NAME)).iterator();
+                assertThat(itr.next().getStatus(), equalTo(SERVING));
+
+                assertThat(service.setStatus(OVERALL_SERVICE_NAME, NOT_SERVING), equalTo(true));
+                assertThat(itr.next().getStatus(), equalTo(NOT_SERVING));
+            }
+        }
+    }
+
+    @Test
+    void clearWatch() throws Exception {
+        DefaultHealthService service = new DefaultHealthService();
+        try (ServerContext serverCtx = GrpcServers.forAddress(localAddress(0)).listenAndAwait(service)) {
+            try (Health.BlockingHealthClient client = GrpcClients.forResolvedAddress(
+                    (InetSocketAddress) serverCtx.listenAddress()).buildBlocking(new Health.ClientFactory())) {
+                assertThat(service.clearStatus(OVERALL_SERVICE_NAME), equalTo(true));
+                BlockingIterator<HealthCheckResponse> itr = client.watch(newRequest(OVERALL_SERVICE_NAME)).iterator();
+                assertThat(itr.next().getStatus(), equalTo(SERVICE_UNKNOWN));
+
+                assertThat(service.setStatus(OVERALL_SERVICE_NAME, SERVING), equalTo(true));
+                assertThat(itr.next().getStatus(), equalTo(SERVING));
+            }
+        }
+    }
+
+    @Test
+    void terminateWatchCheck() throws Exception {
+        DefaultHealthService service = new DefaultHealthService();
+        try (ServerContext serverCtx = GrpcServers.forAddress(localAddress(0)).listenAndAwait(service)) {
+            try (Health.BlockingHealthClient client = GrpcClients.forResolvedAddress(
+                    (InetSocketAddress) serverCtx.listenAddress()).buildBlocking(new Health.ClientFactory())) {
+                assertThat(service.terminate(), equalTo(true));
+                BlockingIterator<HealthCheckResponse> itr = client.watch(newRequest(OVERALL_SERVICE_NAME)).iterator();
+                assertThat(itr.next().getStatus(), equalTo(NOT_SERVING));
+
+                assertThat(service.setStatus(OVERALL_SERVICE_NAME, SERVING), equalTo(false));
+                assertThat(service.clearStatus(OVERALL_SERVICE_NAME), equalTo(false));
+                assertThat(assertThrows(GrpcStatusException.class,
+                                () -> client.check(newRequest(OVERALL_SERVICE_NAME))).status().code(),
+                        equalTo(NOT_FOUND));
+            }
+        }
+    }
+
+    @Test
+    void watchPredicateFails() throws Exception {
```

Comment:
Perhaps `watchPredicateFalse` and another test which throws from the predicate for `Fail`s?